### PR TITLE
feat: Adds tutorial/guide to installing UnderStack on Ubuntu and K3s

### PR DESCRIPTION
**Pull request by nicholaskuechler**
Adds a step by step guide to getting UnderStack up and running on a Ubuntu server with K3s.

### CODE REVIEW

File: docs/install-understack-ubuntu-k3s.md [hunk 1]
```diff
@@ -0,0 +1,305 @@
+# Installing UnderStack on Ubuntu 22.04 + K3s
+
+## Install Pre-requisites
+
+Install some packages we'll need later and some useful troubleshooting utilities.
```
Comment (skrobul) [inline docs/install-understack-ubuntu-k3s.md:5]: are we installing full golang just to have the `go env GOOS` and `go env GOARCH` available?
If so, I think we can safely assume that Ubuntu is `linux` and we can get architecture with `uname -m` although I don't think we will be running ARM servers that soon.

Comment (nicholaskuechler) [reply docs/install-understack-ubuntu-k3s.md:5]: Yeah, I was following their docs which used it and I wasn't sure if I would need it for anything else. I've updated it to remove the go dependency. 👍

File: docs/install-understack-ubuntu-k3s.md [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+## Install Helm
+
+The K3s installer will install kubectl, but we'll also need helm for the UnderStack install.
+
+```bash
+curl https://baltocdn.com/helm/signing.asc | gpg --dearmor | sudo tee /usr/share/keyrings/helm.gpg > /dev/null
+sudo apt-get install apt-transport-https --yes
+echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/helm.gpg] https://baltocdn.com/helm/stable/debian/ all main" | sudo tee /etc/apt/sources.list.d/helm-stable-debian.list
+sudo apt-get update
+sudo apt-get install helm
```
Comment (cardoe) [inline docs/install-understack-ubuntu-k3s.md:36]: Is this the official helm repo?

File: docs/install-understack-ubuntu-k3s.md [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+```
+
+References:
+
+* [https://github.com/bitnami-labs/sealed-secrets?tab=readme-ov-file#installation](https://github.com/bitnami-labs/sealed-secrets?tab=readme-ov-file#installation)
+
+## Install Cert Manager cmctl
+
+```bash
+OS=$(go env GOOS); ARCH=$(go env GOARCH); curl -fsSL -o cmctl.tar.gz https://github.com/cert-manager/cert-manager/releases/latest/download/cmctl-$OS-$ARCH.tar.gz
```
Comment (cardoe) [inline docs/install-understack-ubuntu-k3s.md]: `uname` can replace these.

**Review by cardoe — COMMENTED** [review]
I'm thinking we simplify the install of the binaries with a script.

File: docs/install-understack-ubuntu-k3s.md [hunk 4]
```diff
@@ -0,0 +1,305 @@
+# Installing UnderStack on Ubuntu 22.04 + K3s
+
+## Install Pre-requisites
+
+Install some packages we'll need later and some useful troubleshooting utilities.
+```bash
+apt-get -y install curl jq yq net-tools telnet git apt-transport-https golang-go
```
[the remark this answers is not in the document]
Comment (nicholaskuechler) [reply docs/install-understack-ubuntu-k3s.md]: Already updated 👍

File: docs/install-understack-ubuntu-k3s.md (hunk 2 above)
Comment (nicholaskuechler) [reply docs/install-understack-ubuntu-k3s.md:36]: Yes, I copy/pasted it from: https://helm.sh/docs/intro/install/#from-apt-debianubuntu

File: docs/install-understack-ubuntu-k3s.md (hunk 3 above)
Comment (nicholaskuechler) [reply docs/install-understack-ubuntu-k3s.md]: Already updated 👍

**Comment by nicholaskuechler** [comment]
> I'm thinking we simplify the install of the binaries with a script.

Just the CLI tools like helm, kustomize, kubeseal, etc or a script that installs everything including the k3s and apt package requirements - kind of a "quick-start.sh" that does it all for you?

**Comment by skrobul** [comment]
>     I'm thinking we simplify the install of the binaries with a script.

Just a thought:

**one-time [installation](https://nixos.org/download.html) of nix**
```
sh <(curl -L https://nixos.org/nix/install) --no-daemon
```

**ongoing usage**

```
# on clean system
$ k3s 
bash: k3s: command not found...
$ helm 
bash: k3s: command not found...
$
$ nix-shell
❯ nix-shell

[nix-shell:~/devel/understack]$ k3s
NAME:
   k3s - Kubernetes, but small and simple
```


Apart from having the versioned, stable environment that always works [it is declarative](https://github.com/rackerlabs/understack/blob/main/shell.nix) and does not pollute main system. 
Produces same results on MacOS, Linux and possibly Windows ;)

**Comment by cardoe** [comment]
> > ```
> > I'm thinking we simplify the install of the binaries with a script.
> > ```
> 
> Just a thought:
> 
> **one-time [installation](https://nixos.org/download.html) of nix**
> 
> ... snip ...
> 
> Apart from having the versioned, stable environment that always works [it is declarative](https://github.com/rackerlabs/understack/blob/main/shell.nix) and does not pollute main system. Produces same results on MacOS, Linux and possibly Windows ;)

Yeah I'm with you on that. Realistically virtually all of these things shouldn't have to install these on the actual host as well.

File: components/13-ironic/aio-values.yaml [hunk 5]
```diff
@@ -47,12 +47,42 @@ conf:
...
+endpoints:
+  oslo_messaging:
+    namespace: null
+    auth:
+      admin:
+        username: rabbitmq
+        password: password
+        secret:
+          tls:
+            internal: rabbitmq-tls-direct
```
Comment (cardoe) [inline components/13-ironic/aio-values.yaml:67]: This chunk shouldn't be necessary. We don't use the admin cred.

File: components/13-ironic/aio-values.yaml [hunk 6]
```diff
@@ -47,12 +47,42 @@ conf:
...
+      ironic:
+        username: ironic
+        password: password
+    statefulset:
+      replicas: 3
+      name: rabbitmq-server
+    hosts:
+      default: rabbitmq-nodes
+    host_fqdn_override:
+      default: null
```
Comment (cardoe) [inline components/13-ironic/aio-values.yaml:77]: This isn't necessary cause this is the default when you set `statefulset` and `hosts` like you did above.

File: components/13-ironic/aio-values.yaml [hunk 7]
```diff
@@ -47,12 +47,42 @@ conf:
...
+    path: /ironic
+    scheme: rabbit
+    port:
+      amqp:
+        default: 5672
+      http:
+        default: 15672
```
Comment (cardoe) [inline components/13-ironic/aio-values.yaml:84]: This chunk is the default as well.

File: components/13-ironic/aio-values.yaml [hunk 8]
```diff
@@ -47,12 +47,42 @@ conf:
...
+      rabbit_ha_queues: true
   logging:
     logger_root:
       level: DEBUG
     logger_ironic:
       level: DEBUG
 
+endpoints:
+  oslo_messaging:
+    namespace: null
```
Comment (cardoe) [inline components/13-ironic/aio-values.yaml:60]: This came from the genestack config file from keystone right? Cause that's actually wrong and been planning on deleting it.

Comment (nicholaskuechler) [reply components/13-ironic/aio-values.yaml:60]: This came from understack keystone aio-values: https://github.com/rackerlabs/understack/blob/main/components/10-keystone/aio-values.yaml#L983-L1008

Comment (nicholaskuechler) [reply components/13-ironic/aio-values.yaml:60]: I tried changing it all to just:
```
endpoints:
  oslo_messaging:
    hosts:
      default: rabbitmq-nodes
```
But it doesn't work...

Comment (nicholaskuechler) [reply components/13-ironic/aio-values.yaml:60]: This one works:
```
endpoints:
  oslo_messaging:
    namespace: null
    statefulset:
      name: rabbitmq-server
    hosts:
      default: rabbitmq-nodes
```

File: components/13-ironic/aio-values.yaml [hunk 9]
```diff
@@ -20,6 +20,21 @@ conf:
       automated_clean: false
     dhcp:
       dhcp_provider: none
+    oslo_messaging_rabbit:
+      rabbit_ha_queues: true
+  logging:
+    logger_root:
+      level: DEBUG
+    logger_ironic:
+      level: DEBUG
```
Comment (cardoe) [inline components/13-ironic/aio-values.yaml:29]: So I think the logging chunk can be dropped.

**Review by cardoe — APPROVED** [review]

**Pull request merged**